src/mame/drivers/starfire.cpp

```diff
@@ -213,8 +201,8 @@ uint8_t fireone_state::input_r(offs_t offset)
 			return m_system->read();
 		case 2:
 		{
-			const uint8_t input = m_controls[m_player_select]->read();
-			return (input & 0xc0) | s_paddle_map[input & 0x3f];
+			uint8_t input = m_controls[m_player_select]->read();
+			return input ^ BIT(input, 1, 5); // paddle portion is a 6-bit Gray code
```

Comment:
Why did you get rid of the `const`?  The local `input` shouldn’t be changed after initialisation.